tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolAtCursorTest.java

```diff
@@ -227,4 +233,31 @@ public void testQuotedIdentifiers(int line, int column, String expSymbolName) {
                 {88, 9, "'anydata"}
         };
     }
+
+    @Test(dataProvider = "SymWithErrorPosProvider")
+    public void testVarSymbolsWithCompileErrorType(int line, int col, TypeDescKind typeKind, DiagnosticState state) {
+        Project project = BCompileUtil.loadProject("test-src/var_symbols_with_error_type_test.bal");
+        SemanticModel model = getDefaultModulesSemanticModel(project);
+        Document srcFile = getDocumentForSingleSource(project);
+
+        Optional<Symbol> symbol = model.symbol(srcFile, LinePosition.from(line, col));
+
+        assertEquals(symbol.get().kind(), VARIABLE);
+        assertEquals(((VariableSymbol) symbol.get()).typeDescriptor().typeKind(), typeKind);
+        assertEquals(((VariableSymbol) symbol.get()).diagnosticState(), state);
+    }
+
+    @DataProvider(name = "SymWithErrorPosProvider")
+    public Object[][] getSymWithErrPos() {
+        return new Object[][]{
+                {17, 8, INT, DiagnosticState.VALID},
+                {20, 10, COMPILATION_ERROR, DiagnosticState.REDECLARED},
+                {23, 8, COMPILATION_ERROR, DiagnosticState.FAILED_TO_DETERMINE_TYPE},
+                {24, 8, COMPILATION_ERROR, DiagnosticState.FAILED_TO_DETERMINE_TYPE},
+                {25, 8, COMPILATION_ERROR, DiagnosticState.FAILED_TO_DETERMINE_TYPE},
+                {27, 8, COMPILATION_ERROR, DiagnosticState.FAILED_TO_DETERMINE_TYPE},
```

Comment:
Better if a test case is added to determine the DiagnosticState of `Person person = new;` which turns out to be valid.